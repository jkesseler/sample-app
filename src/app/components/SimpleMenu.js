/**
  TODO: Refactor the menu
  Maybe get routerConfig from state?
*/
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Menu from '@material-ui/core/Menu';
import IconButton from '@material-ui/core/IconButton';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/Menu';


export class SimpleMenu extends Component {
  constructor(props) {
    super(props);
    this.state = { isOpen: false };
  }

  handleToggleMenu = () => {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  };

  handleCloseMenu = () => {
    this.setState({ isOpen: false });
  };

  render() {
    const { menuItems } = this.props;
    const { isOpen } = this.state;

    return (
      <>
        <IconButton
          color="inherit"
          aria-label="Menu"
          onClick={this.handleToggleMenu}
        >
          <MenuIcon />
        </IconButton>
        <Menu
          id="simple-menu"
          open={isOpen}
          onClose={this.handleCloseMenu}
          style={{ marginTop: 16 }}
        >
          {menuItems.map(item => (
            <Link to={item.path} key={item.path} style={{ textDecoration: 'none', display: 'block' }}>
              <MenuItem>
                {item.title}
              </MenuItem>
            </Link>
          ))}
        </Menu>
      </>
    );
  }
}

SimpleMenu.propTypes = {
  menuItems: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default SimpleMenu;
