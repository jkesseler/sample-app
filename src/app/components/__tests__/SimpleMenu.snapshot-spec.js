import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SimpleMenu from '../SimpleMenu';

const defaultProps = {
  menuItems: [{
    path: '/',
    title: 'test',
  }],
};

describe('SimpleMenu', () => {
  it('renders without crashing', () => {
    const component = shallow(<SimpleMenu {...defaultProps} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
