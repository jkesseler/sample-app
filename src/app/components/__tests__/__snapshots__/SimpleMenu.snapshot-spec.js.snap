// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleMenu renders without crashing 1`] = `
<Fragment>
  <WithStyles(IconButton)
    aria-haspopup="true"
    aria-label="Menu"
    color="inherit"
    onClick={[Function]}
  >
    <pure(MenuIcon) />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    id="simple-menu"
    onClose={[Function]}
    open={false}
    style={
      Object {
        "marginTop": 16,
      }
    }
  >
    <Link
      key="/"
      replace={false}
      style={
        Object {
          "display": "block",
          "textDecoration": "none",
        }
      }
      to="/"
    >
      <WithStyles(MenuItem)
        onClick={[Function]}
      >
        test
      </WithStyles(MenuItem)>
    </Link>
  </WithStyles(Menu)>
</Fragment>
`;
